### PR TITLE
Preliminary changes to python error handling

### PR DESCRIPTION
**Pull request by BrentTanKian**
Not 100 percent sure if this is right. Do let me know if there's anything wrong or to be improved on so I can make the appropriate changes

### CODE REVIEW

File: py/src/main.py [hunk 1]
```diff
@@ -15,11 +18,19 @@ def main(argv: List[str]) -> int:
     try:
         gameboy = GameBoy(args)
         gameboy.run()
+    except (InvalidOpcode, InvalidRamRead, InvalidRamWrite) as e:
+        raise GameException(e) 
+    except (Quit, Timeout, UnitTestFailed, UnitTestPassed) as e:
+        raise ControlledExit(e)
+    except (RomMissing, LogoChecksumFailed, HeaderChecksumFailed) as e:
+        raise UserException(e)
```
Comment (shish) [inline py/src/main.py:26]: `InvalidOpcode` is already a sub-class of `GameException`, no need to catch the specific type and raise the generic type (and the same for the others - ie, these highlighted lines don't really add anything, so they can go)

**Review by shish — CHANGES_REQUESTED** [review]
The organisation of the errors looks good - the extra `except (...)` parts in `main.py` are redundant though

File: py/src/main.py (hunk 1 above)
Comment (BrentTanKian) [reply py/src/main.py:26]: Hey sure thing, I'll remove those lines and push up again

**Comment by shish** [comment]
Looks good, thank you :D

**Comment by shish** [comment]
(Derp, should have waited for the automated tests ^^ Fixed now)

**Pull request merged**